jcli/src/jcli_app/certificate/new_vote_cast.rs

```diff
@@ -3,42 +3,94 @@ use chain_impl_mockchain::{
     certificate::{Certificate, VoteCast, VotePlanId},
     vote::{Choice, Payload},
 };
+use rand_chacha::rand_core::SeedableRng;
 use std::path::PathBuf;
 use structopt::StructOpt;
 
-/// create a vote cast certificate
 #[derive(StructOpt)]
-pub struct VoteCastCmd {
+pub struct PublicVoteCast {
+    /// the vote plan identified on the blockchain
+    vote_plan_id: VotePlanId,
+
+    /// the number of proposal in the vote plan you vote for
+    proposal_index: u8,
+
+    /// the number of choice within the proposal you vote for
+    choice: u8,
+
+    /// write the output to the given file or print it to the standard output if not defined
+    #[structopt(long = "output")]
+    output: Option<PathBuf>,
+}
+
+#[derive(StructOpt)]
+pub struct PrivateVoteCast {
     /// the vote plan identified on the blockchain
-    pub vote_plan_id: VotePlanId,
+    vote_plan_id: VotePlanId,
 
     /// the number of proposal in the vote plan you vote for
-    pub proposal_index: u8,
+    proposal_index: u8,
+
+    /// size of voting options
+    options: usize,
```

Comment:
Should this be retrievable from the proposal status endpoint in REST?

Reply:
It is though